src/App.jsx

```diff
@@ -14,33 +16,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const FIELDS = {
+  name: 'name',
+  length: 'length',
+  none: 'none',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortField, isReversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField !== FIELDS.none) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case FIELDS.name:
+          return good1.localeCompare(good2);
+        case FIELDS.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(FIELDS.none);
```

Comment:
also you can use only useState(''),  Instead : useState(FIELDS.none);

Reply:
so i dont see problem but ok